src/app/cli/main.cpp

```diff
@@ -76,30 +78,41 @@ static void PrintVersion()
 }
 
 static Interpreter gInterpreter;
+static sigset_t    gSignalSet;
 
-static void HandleSignalInterrupt(int)
+static void HandleSignalInterrupt()
 {
-    gInterpreter.AbortCommand();
+    int signalNum = 0;
+
+    while (true)
+    {
+        sigwait(&gSignalSet, &signalNum);
+        gInterpreter.AbortCommand();
```

Comment:
should we quit this thread after this?

Reply:
This Thread is designed to run in background to handle the interrupt signals. As the user can input `CRTL + C` in the CLI for multiple times, we need to keep this thread running to handle more interrupts (The name `AbortCommand` is misleading, it should be `CancelCommand` and I have fired a PR https://github.com/openthread/ot-commissioner/pull/135 to rename it).

Reply:
nit: If this thread goes on, you need to make sure that `AbortCommand` is reentrant.

Reply:
Sure, it is.